api-extractor/src/definitions/ApiItem.ts

```diff
@@ -347,23 +351,23 @@ abstract class ApiItem {
    * an \@inheritdoc referencing ApiItemTwo, and ApiItemTwo has an \@inheritdoc refercing ApiItemOne then
    * we have a circular dependency and an error will be reported.
    */
-  public resolveReferences(): void {
+  public canResolveReferences(): boolean {
     switch (this._state) {
       case ResolveState.Resolved:
-        return;
+        return true;
       case ResolveState.Unresolved:
         this._state = ResolveState.Resolving;
 
         for (const innerItem of this.innerItems) {
-          innerItem.resolveReferences();
+          innerItem.canResolveReferences();
         }
         this.onResolveReferences();
 
         this._state = ResolveState.Resolved;
-        return;
+        return true;
       case ResolveState.Resolving:
         this.reportError('circular reference');
-        return;
+        return false;
```

Comment:
Suggest to rename this function to tryResolveReferences().  We should audit every caller to distinguish between:

1. The general "phase 2" visitor that's trying to make sure everything ends up resolved
2. A link or inheritdoc that's counting on resolveReferences() to put the item in a good state (e.g. to know if it's @public or to copy its summary); if it returns false, they need to skip whatever they were trying to do.